use-cases/client-server-msgs-perf/svmsg_file_server.c

```diff
@@ -543,22 +615,26 @@ svr_op_ct_init(requestMsg *req)
     resp.clientId = req->clientId;   // For this client
 
     // Use the next free slot w/o trying to recycle existing free slots.
-    resp.client_idx = NumActiveClientsHWM;
+    resp.client_idx = NumActiveClientsHWM++;
```

Comment:
Atomically increment `NumActiveClientsHWM` and then use old-value of `NumActiveClientsHWM` as this client's index .

Reply:
I think this is wrong for C code. I don't know C++ but assume the ++ operator will do the right thing, but for C code you want something like `__sync_fetch_and_add()`. To keep things simpler you might want to just use that throughout and not bother the C++ atomics stuff. (I don't feel strongly either way on that though, just a thought.)

Reply:
`__sync_fetch_and_add()` was the earlier interface, but more recently, standard 'C' has also supported use of `Atomic` variables that allow for this kind of `++ ` or `--` avoiding races.

I did some research in this before choosing this approach:

See [this stackoverflow post](https://stackoverflow.com/questions/59662263/making-data-reads-writes-atomic-in-c11-gcc-using-stdatomic-h).

From [cpp-reference on atomic](https://en.cppreference.com/w/c/language/atomic), I see this:

```
The header <stdatomic.h> defines [37 convenience type aliases](https://en.cppreference.com/w/c/thread#Atomic_operations), from [atomic_bool](https://en.cppreference.com/w/c/thread#Atomic_operations) to [atomic_uintmax_t](https://en.cppreference.com/w/c/thread#Atomic_operations), which simplify the use of this keyword with built-in and library types.

_Atomic const int * p1;  // p is a pointer to an atomic const int
const [atomic_int](http://en.cppreference.com/w/c/atomic) * p2;   // same
const _Atomic(int) * p3; // same
```

From [C thread-concurrency pages](https://en.cppreference.com/w/c/thread), I see this:

```
Typedef name 	Full type name
atomic_bool(C11) 	_Atomic _Bool
atomic_char(C11) 	_Atomic char
atomic_schar(C11) 	_Atomic signed char
atomic_uchar(C11) 	_Atomic unsigned char
atomic_short(C11) 	_Atomic short
atomic_ushort(C11) 	_Atomic unsigned short
atomic_int(C11) 	_Atomic int
```

I think for our limited use of 64-bit counters, this use of `_Atomic` should be fine.

Comments welcome.